src/bioclip_vector_db/vector_db.py

```diff
@@ -0,0 +1,187 @@
+"""
+Author: Sreejith Menon 
+Executable script for setting up a database of vectors from the bioclip dataset
+"""
+import argparse
+import chromadb
+import datasets
+import enum
+import logging
+import os
+import torch 
+
+from bioclip.predict import TreeOfLifeClassifier
+from tqdm import tqdm
+from typing import List
+
+_DEFAULT_OUTPUT_DIR = os.path.join(os.getcwd(), 
+                                   "vector_db")
+_LOG_FORMAT = "[%(asctime)s] [%(levelname)s] [%(name)s] %(message)s"
+logging.basicConfig(level=logging.INFO, format=_LOG_FORMAT)
+logger = logging.getLogger()
+
+def _get_device() -> torch.device:
+    if torch.cuda.is_available():
+        logger.info("CUDA is available")
+        return torch.device("cuda")
+    elif torch.mps.is_available():
+        logger.info("MPS is available")
+        return torch.device("mps")
+    else:
+        logger.warning("CUDA and MPS are not available. Default to CPU")
+        return torch.device("cpu")
+    
+class HfDatasetType(enum.Enum):
+    BIRD = "Somnath01/Birds_Species"
+    TREE_OF_LIFE = "imageomics/TreeOfLife-10M"
+
+class BioclipVectorDatabase: 
+    def __init__(self, dataset_type: HfDatasetType, 
+                 collection_dir: str, 
+                 split: str):
+        self._dataset_type = dataset_type
+        self._classifier = TreeOfLifeClassifier(
+            device=_get_device())
+        self._dataset = None
+        self._collection_dir = collection_dir
+        self._client = None
+        self._collection = None
+        
+        self._prepare_dataset(
+            split=split)
+        self._init_collection()
+        
+    def _prepare_dataset(self, 
```

Comment:
Could we add functionality to work alternatively with images on disk as an alternative to HF dataset? The full training set for BioCLIP wasn't able to make it up to HF due to redistribution restrictions.